### DISCUSSION TITLE
Trouble getting started with osgEarthCesium nodekit

### DISCUSSION THREAD

**Discussion by simbu11** [pelicanmapping/osgearth#2384] [Support Q&A]
I was keen to give the new osgEarthCesium nodekit a spin, but have run into a problem getting going.

I'm on Windows 11, Visual Studio 2022, NVIDIA GeForce RTX 4060. I used vcpkg for installing dependencies (osg etc).

I cloned the osgEarth and cesium-native repos (Nov 17th for both) and managed to build cesium-native ok. For osgEarth I needed to update a few of the linked library names in the CMake config/VS project, as cesium-native has obviously done some slight reorganization vs osgEarth master. However, I managed to successfully build the osgEarth and the nodekit (osgEarthCesium.dll is sitting in my bin folder). I could successfully run some of the osgEarth test files in e.g. osgearth_viewer (e.g. arcgisonline, readymap etc)

I wanted to load a public 3DTiles as [outlined here](https://docs.osgearth.org/en/latest/cesium_native.html). However, when I run a simple example .earth file:

```
<map name ="Test">
	<CesiumNative3DTiles name="agi">
	    <url>https://pelican-public.s3.amazonaws.com/3dtiles/agi-hq/tileset.json</url>
	</CesiumNative3DTiles>
</map>
```

I get a warning:
`[osgearth info] [EarthSerializer2] Failed to find an extension for cesiumnative3dtiles_`

and only the white earth is shown. I stepped into EarthFileSerializer2.cpp and Extension* loadExtension(const Config& conf), where it fails to load the extension/driver.

I'm not sure if it is due to a missing file in my bin folder, something missing from my PATH, or just something wrong with the .earth file itself (it could be really nice to provide a 3dtiles.earth file to eliminate the latter at least!).

Would be great to hear if you have any hints!

**Comment by jasonbeverage** [comment]
It is probably just the fact that your osgEarthCesium library isn't loaded by default.  Try either passing in -l osgEarthCesium to your osgearth_viewer command line like:
```
osgearth_viewer -l osgEarthCesium mymap.earth
```

Or you can do the same thing by adding <libraries>osgEarthCesium</libraries> to your earth file like this
```
<map name ="Test">
        <libraries>osgEarthCesium</libraries>
	<CesiumNative3DTiles name="agi">
	    <url>https://pelican-public.s3.amazonaws.com/3dtiles/agi-hq/tileset.json</url>
	</CesiumNative3DTiles>
</map>
```

**Comment by simbu11** [comment]
Hi Jason,

Many thanks, that was exactly the problem. Both of your solutions solved the problem for me, and the 3DTiles data now loads as expected.
